app/uk/gov/hmrc/customs/notification/services/PublicNotificationRequestService.scala

```diff
@@ -18,54 +18,29 @@ package uk.gov.hmrc.customs.notification.services
-import uk.gov.hmrc.customs.notification.controllers.CustomHeaderNames.{X_BADGE_ID_HEADER_NAME, X_CDS_CLIENT_ID_HEADER_NAME, X_CONVERSATION_ID_HEADER_NAME}
+import uk.gov.hmrc.customs.notification.controllers.CustomHeaderNames.X_BADGE_ID_HEADER_NAME
+import uk.gov.hmrc.customs.notification.controllers.RequestMetaData
 import uk.gov.hmrc.customs.notification.domain.{DeclarantCallbackData, Header, PublicNotificationRequest, PublicNotificationRequestBody}
-import uk.gov.hmrc.http.HeaderCarrier
 
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
 import scala.xml.NodeSeq
 
 @Singleton
 class PublicNotificationRequestService @Inject()(apiSubscriptionFieldsConnector: ApiSubscriptionFieldsConnector) {
 
-  def createRequest(xml: NodeSeq, headers: Headers)(implicit hc: HeaderCarrier): Future[Option[PublicNotificationRequest]] = {
-    for {
-      (clientId, conversationId, mayBeBadgeId) <- getHeaders(headers)
-      maybeClientData <- getClientData(clientId)
-      maybePublicNotificationRequest <- maybePublicNotificationRequest(clientId, xml, conversationId, mayBeBadgeId, maybeClientData)
-    } yield {
-      maybePublicNotificationRequest
-    }
-  }
-
-  private def getHeaders(headers: Headers): Future[(String, String, Option[String])] = {
-    // headers have been validated so safe to do a naked get except badgeId which is optional
-    Future.successful((headers.get(X_CDS_CLIENT_ID_HEADER_NAME).get,
-      headers.get(X_CONVERSATION_ID_HEADER_NAME).get,
-      headers.get(X_BADGE_ID_HEADER_NAME).flatMap(x => if (x.isEmpty) None else Some(x)))
-    )
-  }
+  def createRequest(notificationXML: NodeSeq, clientData: DeclarantCallbackData, metaData: RequestMetaData): PublicNotificationRequest = {
 
-  private def getClientData(clientId: String)(implicit hc: HeaderCarrier) = {
-    apiSubscriptionFieldsConnector.getClientData(clientId)
-  }
-
-  // xml.toString() may be processor intensive for large payloads so we do not block
-  private def maybePublicNotificationRequest(clientId: String,
-                                             xml: NodeSeq,
-                                             conversationId: String,
-                                             mayBeBadgeId: Option[String],
-                                             maybeClientData: Option[DeclarantCallbackData]): Future[Option[PublicNotificationRequest]] = {
-    Future {
-      maybeClientData.fold[Option[PublicNotificationRequest]](None) { clientData =>
-        val outboundCallHeaders: Seq[Header] = mayBeBadgeId.fold(Seq[Header]())(x => Seq(Header(X_BADGE_ID_HEADER_NAME, x)))
-        val request = PublicNotificationRequest(clientId, PublicNotificationRequestBody(clientData.callbackUrl, clientData.securityToken, conversationId, outboundCallHeaders, xml.toString()))
-        Some(request)
-      }
+    val outboundCallHeaders: Seq[Header] = metaData.mayBeBadgeId match {
```

Comment:
```Scala
      case x if x == None || x.get.isEmpty => Seq()
      case Some(badgeId) => Seq(Header(X_BADGE_ID_HEADER_NAME, badgeId))
```
Unecessay `if` statement. Consider writing the above as
```
      case Some(badgeId) => Seq(Header(X_BADGE_ID_HEADER_NAME, badgeId))
      case _ => Seq()
```

Reply:
no, it is required. you can test that by making the change and running the unit tests